python/lsst/daf/butler/_quantum_backed.py

```diff
@@ -531,6 +531,53 @@ def retrieve_artifacts_zip(
         """
         return retrieve_and_zip(refs, destination, self._datastore.retrieveArtifacts, overwrite)
 
+    def retrieveArtifacts(
```

Comment:
If we want to eventually make all methods snake-case, maybe call it `retrieve_artifacts` instead? I guess this is for consistency with `Butler`, but `LimitedButler` does not have the abstract method for this?

Reply:
It seems vaguely confusing to me to have `QBB.retrieve_artifacts` and `Butler.retrieveArtifacts`. @TallJimbo ?

Reply:
Let's add the snake case version everywhere, even if we don't drop the camelCase one.

Reply:
I'll do that on a different ticket since I think I need to think about it a little before adding all the butler alternative APIs with different parameters.